lib/slack-hook-handler.js

```diff
@@ -31,22 +33,120 @@ SlackHookHandler.prototype.getIntent = function(slackUser) {
+    var roomID = this.rooms.matrixRoomID(params.channel_id);
+    if (params.text) {
+        intent.sendText(roomID, substitutions.slackToMatrix(params.text));
+        return;
+    }
+    this.lookupAndSendNonTextMessage(params.channel_id, params.timestamp, intent, roomID);
+    return;
+};
+
+/**
+ * Attempts to handle a non-text message received from a slack webhook request.
+ *
+ * @param {string} channelID Slack channel ID.
+ * @param {string} timestamp Timestamp when message was received, in seconds
+ *     formatted as a float.
+ * @param {Intent} intent Intent for sending messages as the relevant user.
+ * @param {string} roomID Matrix room ID associated with channelID.
+ */
+SlackHookHandler.prototype.lookupAndSendNonTextMessage =
+        function(channelID, timestamp, intent, roomID) {
+    if (!this.config.slack_master_token) {
+        return;
+    }
+    var params = {
+        channel: channelID,
+        latest: timestamp,
+        oldest: timestamp,
+        inclusive: "1",
+        token: this.config.slack_master_token,
+    };
+    var self = this;
+    this.requestLib.post("https://slack.com/api/channels.history", {form: params},
+            function(err, _, body) {
+        if (err) {
+            console.log("Error looking up history: %s", err);
+        }
+        else {
+            var resp = JSON.parse(body);
+            if (!resp.ok || !resp.messages) {
+                console.log("Could not find history: " + body);
+                return;
+            }
+            if (resp.messages.length != 1) {
+                console.log("Really unlucky, got multiple messages at same" +
+                    " microsecond, dropping:" + body);
+            }
+            var message = resp.messages[0];
+            if (!message.file) {
+                console.log("Ignoring non-text non-image message: " + body);
+                return;
             }
-            else {
-                console.log(
-                    "Ignoring message for slack channel " +
-                    "with unknown matrix ID: " + params.channel_id +
-                    " (" + params.channel_name + ")"
-                );
+            if (message.file.mimetype && message.file.mimetype.indexOf("image/") === 0) {
+                var matrixMessage = self.slackImageToMatrixImage(message.file);
+                intent.sendMessage(roomID, matrixMessage);
+                if (message.file.initial_comment) {
+                    var text = substitutions.slackToMatrix(
+                        message.file.initial_comment.comment
+                    );
+                    intent.sendText(roomID, text);
+                }
+                return;
             }
+            console.log("Ignoring non-image attachment: " + body);
+        }
+    });
+};
+
+/**
+ * Converts a slack image attachment to a matrix image event.
+ *
+ * @param {Object} file The slack image attachment file object.
+ * @return {Object} Matrix event content.
```

Comment:
Docs on `@param` would be cool. Sadly it isn't actually valid JSDoc for the reverse (doccing properties on `@return`), but you can always `@typedef` it e.g. `@return {Foo} A foo` and then have:
```javascript
/**
 * @typedef Foo
 * @property {string} bar A bar
 */
```